MM-30507. Added assign client classes to superscopes.

rachelbritton opened the pull request:
Added section on assigning classes to superscopes and a link from the Kea DHCP page.

rachelbritton committed:
MM-30507. Added assign client classes to superscopes.

Added section on assigning classes to superscopes and a link from the DHCP Kea page.

rachelbritton committed:
MM-30507. Added client class API info.

Updated Managing Kea Client Classification docs and added info about API commands, including GetClientClass.

rachelbritton commented:
@ethorlacius, @fbrcic-bc provided documentation on GetClientClass and the other commands that can be used in relation to client classes. I've added a section to the documentation here, as we had not documented them at all (not only for superscopes).

ethorlacius commented on guides/admin-manual/admin_dhcp_kea.rst:
```diff
@@ -266,3 +272,218 @@ Assigning client classes to scopes/pools shows up in the history of the respecti
+
+**AddClientClass**
+
+The XML command ``AddClientClass`` is used to add/define a new client class on a Kea server.
```
Unrelated to this PR, but are we using XML in all our API examples? We really should switch to JSON.  XML is old and uncool, but JSON is modern and readable

rachelbritton replied:
@fbrcic-bc

ethorlacius replied:
But are there other API examples in the docs? If not, we shouldn't add these here.  If there are, do they use XML? Then we should keep this as is but make an issue to bulk convert them to JSON at some point

fbrcic-bc replied:
I saw that from this example: https://docs.menandmice.com/en/latest/guides/admin-manual/admin_subnet_monitoring_events/

But these examples could be transformed into JSON, if you think that would be better, and it probably would.

rachelbritton replied:
@ethorlacius, Do you want me to remove the examples for now, then? Until they are transformed into JSON?

ethorlacius commented on guides/admin-manual/admin_dhcp_kea.rst:
```diff
@@ -211,8 +215,10 @@ You can manage Kea Client Classifications through Micetro.
       :width: 70%
       
-   * If you have any client classes already defined on your server, you can find them listed on the respective service type tab (DHCPv4/DHCPv6). 
+   * If you have any client classes already defined on your server, you can find them listed on the respective service type tab (DHCPv4/DHCPv6). Here, you can see whether existing client classes are built-in, global (i.e., replicated on all available Kea servers), and/or custom.
```
The "global" flag is unique to Micetro. A Kea admin knows what client classes are, what the difference is between built-in and custom etc, but "global" might require a special mention.  Something like
"If a client class is marked as Global, it will be replicated to all Kea servers managed by Micetro"  ?

rachelbritton committed:
MM-30507. Removed XML examples.

Removed XML examples, as it was determined that they should be JSON if we are going to include them.
